osv/ecosystems.py

```diff
@@ -76,7 +76,7 @@ def _before_limits(self, version, limits):
   def next_version(self, package, version):
     """Get the next version after the given version."""
     versions = self.enumerate_versions(package, version, fixed=None)
-    if versions and versions[0] != version:
+    if versions and self.sort_key(versions[0]) != self.sort_key(version):
```

Comment:
Can you explain a bit more what this was running into before? Good to add a comment here to add more context for future readers.